packages/typespec-ts/src/utils/modelUtils.ts

```diff
@@ -1255,3 +1247,20 @@ export function trimUsage(model: any) {
   }, {});
   return ordered;
 }
+
+export function isAzureCoreErrorType(t?: Type): boolean {
+  if (
+    t?.kind !== "Model" ||
+    !["Error", "ErrorResponse", "InnerError", "ErrorModel"].includes(t.name)
```

Comment:
shall we consider this comparison to be case-insensitive?  I double whether people would write sth. like 'Errormodel'